### PR TITLE
Add mpd_(send|run)_add_id_whence functions to support the new whence …

### PR DESCRIPTION
**Pull request by jcorporation**
…parameter for addid

### CODE REVIEW

File: src/queue.c [hunk 1]
```diff
@@ -195,6 +195,30 @@ mpd_send_add_id_to(struct mpd_connection *connection, const char *uri,
...
+		whence_s = "+";
+		break;
+
+	case MPD_POSITION_BEFORE_CURRENT:
+		whence_s = "-";
+		break;
+	}
+
+	char to_str[64] = "";
+	snprintf(to_str, 64, " position %s%u", whence_s, to);
```
Comment (MaxKellermann) [inline src/queue.c:217]: This is not the right syntax for `addid`!

Comment (jcorporation) [reply src/queue.c]: sorry, copy and paste error from search code...
fixed with the last commit and I also centraliced the whence char handling

File: include/mpd/queue.h [hunk 2]
```diff
@@ -287,6 +302,20 @@ int
...
 		  unsigned to);
 
+/**
+ * Executes the "addid" command and reads the response.
+ * file is always a single file or URL.
+ *
+ * @param connection the connection to MPD
+ * @param uri the URI of the song to be added
+ * @param to the desired position of the song
+ * @param whence how to interpret the position parameter
```
Comment (MaxKellermann) [inline include/mpd/queue.h:312]: There is no `position` parameter

File: src/queue.c [hunk 3]
```diff
@@ -195,6 +195,18 @@ mpd_send_add_id_to(struct mpd_connection *connection, const char *uri,
 	return mpd_send_s_u_command(connection, "addid", uri, to);
 }
 
+bool
+mpd_send_add_id_whence(struct mpd_connection *connection, const char *uri,
+		   unsigned to, enum mpd_position_whence whence)
+{
+	const char *whence_s = mpd_position_whence_char(whence);
+
+	char to_str[64] = "";
```
Comment (MaxKellermann) [inline src/queue.c:204]: What's the point of the initializer? It will be overwritten in the next line, so this initializer just adds pointless overhead.

**Pull request merged**